rust/src/metrics/mod.rs

```diff
@@ -295,6 +295,11 @@ impl<D: Domain> MetricSpace for (VectorDomain<D>, HammingDistance) {
         })
     }
 }
+impl MetricSpace for (BitVectorDomain, HammingDistance) {
+    fn check_space(&self) -> Fallible<()> {
+        Ok(())
+    }
+}
```

Comment:
While this is true, if you end up not using it, then it can be removed. Similarly for (VectorDomain, DiscreteDistance) below.